implants/eldritch/src/file.rs

```diff
@@ -17,24 +17,34 @@ mod write_impl;
 use derive_more::Display;
 
 use starlark::environment::{Methods, MethodsBuilder, MethodsStatic};
-use starlark::values::{StarlarkValue, Value, UnpackValue, ValueLike};
 use starlark::values::none::NoneType;
+use starlark::values::{StarlarkValue, Value, UnpackValue, ValueLike, ProvidesStaticType};
 use starlark::{starlark_type, starlark_simple_value, starlark_module};
+use serde::{Serialize,Serializer};
 
-#[derive(Copy, Clone, Debug, PartialEq, Display)]
+#[derive(Copy, Clone, Debug, PartialEq, Display, ProvidesStaticType)]
 #[display(fmt = "FileLibrary")]
 pub struct FileLibrary();
 starlark_simple_value!(FileLibrary);
 
 impl<'v> StarlarkValue<'v> for FileLibrary {
     starlark_type!("file_library");
 
-    fn get_methods(&self) -> Option<&'static Methods> {
+    fn get_methods() -> Option<&'static Methods> {
```

Comment:
Compiler error thrown when upgraded to 0.9.0 this seems to be the new pattern.

Reply:
![image](https://user-images.githubusercontent.com/7121375/200078276-d31ac08c-f525-4ffa-ab1b-8c20e067cebc.png)